test/RefundProvider.ts

```diff
@@ -0,0 +1,52 @@
+import { MockVaultManager } from "../typechain-types"
+import { DealProvider, IDealProvierEvents } from "../typechain-types/contracts/DealProvider"
+import { LockDealNFT } from "../typechain-types/contracts/LockDealNFT"
+import { RefundProvider } from "../typechain-types/contracts/RefundProvider/RefundProvider.sol"
+import { ERC20Token } from "../typechain-types/poolz-helper-v2/contracts/token"
+import { deployed } from "./helper"
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers"
+import { expect } from "chai"
+import { constants, BigNumber } from "ethers"
+
+describe("Refund Provider", function () {
+    let dealProvider: DealProvider
+    let refundProvider: RefundProvider
+    let lockDealNFT: LockDealNFT
+    let poolId: number
+    let receiver: SignerWithAddress
+    let newOwner: SignerWithAddress
+    let token: ERC20Token
+    let BUSD: ERC20Token
+    let params: [number, number, number]
+    let poolData: [IDealProvierEvents.BasePoolInfoStructOutput, BigNumber[]] & {
+        poolInfo: IDealProvierEvents.BasePoolInfoStructOutput
+        params: BigNumber[]
+    }
+    const amount = 10000
+
+    before(async () => {
+        [receiver, newOwner] = await ethers.getSigners()
+        const mockVaultManager: MockVaultManager = await deployed("MockVaultManager")
+        lockDealNFT = await deployed("LockDealNFT", mockVaultManager.address)
+        token = await deployed("ERC20Token", "TEST Token", "TERC20")
+        BUSD = await deployed("ERC20Token", "BUSD Token", "BUSD")
```

Comment:
lets try to do test without the token.
the token is not a part of this system (the vault will handle this)

we just need to make sure:
1. the state is good after each action
2. the vault called with the right amount / times

Reply:
`LockedDealNFT` checks allowance amount:
https://github.com/The-Poolz/LockDealNFT/blob/e3f1cc8503927e79d74bd2695dda17b1250b073a/contracts/LockDealNFT/LockDealNFTModifiers.sol#L67

We can remove this part.